src/vc.rs

```diff
@@ -537,29 +592,229 @@ impl Credential {
+    pub async fn generate_jwt(
+        &self,
+        jwk: Option<&JWK>,
+        options: &LinkedDataProofOptions,
+    ) -> Result<String, Error> {
+        let LinkedDataProofOptions {
+            verification_method,
+            proof_purpose,
+            created,
+            challenge,
+            domain,
+            checks,
+        } = options;
+        if checks.is_some() {
+            return Err(Error::UnencodableOptionClaim("checks".to_string()));
+        }
+        if created.is_some() {
+            return Err(Error::UnencodableOptionClaim("created".to_string()));
+        }
+        match proof_purpose {
+            None => (),
+            Some(ProofPurpose::AssertionMethod) => (),
+            Some(_) => return Err(Error::UnencodableOptionClaim("proofPurpose".to_string())),
+        }
+        let claims = JWTClaims {
+            nonce: challenge.clone(),
+            audience: match domain {
+                Some(domain) => Some(OneOrMany::One(StringOrURI::try_from(domain.to_string())?)),
+                None => None,
+            },
+            ..self.to_jwt_claims()?
+        };
+        let algorithm = if let Some(jwk) = jwk {
+            jwk.get_algorithm().ok_or(Error::MissingAlgorithm)?
+        } else {
+            crate::jwk::Algorithm::None
+        };
+        let key_id = match (
+            jwk.and_then(|jwk| jwk.key_id.clone()),
+            verification_method.to_owned(),
+        ) {
+            (Some(jwk_kid), None) => Some(jwk_kid),
+            (None, Some(vm_id)) => Some(vm_id),
+            (None, None) => None,
+            (Some(jwk_kid), Some(vm_id)) if jwk_kid == vm_id => Some(vm_id),
+            (Some(jwk_kid), Some(vm_id)) => return Err(Error::KeyIdVMMismatch(vm_id, jwk_kid)),
+        };
+        let header = Header {
+            algorithm,
+            key_id,
+            ..Default::default()
+        };
+        let header_b64 = base64_encode_json(&header)?;
+        let payload_b64 = base64_encode_json(&claims)?;
+        if let Some(jwk) = jwk {
+            let signing_input = header_b64 + "." + &payload_b64;
+            let sig_b64 = crate::jws::sign_bytes_b64(algorithm, signing_input.as_bytes(), jwk)?;
+            let jws = signing_input + "." + &sig_b64;
+            Ok(jws)
+        } else {
+            let jwt = header_b64 + "." + &payload_b64 + ".";
+            Ok(jwt)
+        }
+    }
+
+    pub async fn verify_jwt(
+        jwt: &str,
+        options_opt: Option<LinkedDataProofOptions>,
+        resolver: &dyn DIDResolver,
+    ) -> VerificationResult {
+        let (_vc, result) = Self::decode_verify_jwt(jwt, options_opt, resolver).await;
+        result
+    }
+
+    pub async fn decode_verify_jwt(
+        jwt: &str,
+        options_opt: Option<LinkedDataProofOptions>,
+        resolver: &dyn DIDResolver,
+    ) -> (Option<Self>, VerificationResult) {
+        // let mut options = options_opt.unwrap_or_default();
```

Comment:
leftover comment

Reply:
Fixed in d7747d724798c2f46bb3a2e0749c5470b754d4c8